GitUpdateScript.sh

```diff
@@ -1,122 +1,108 @@
+function MoveToDefaultBranches(){
+    local arrayDefaultBranches=(master develop)
+    for (( x=0; x<${#arrayDefaultBranches[*]}; x++ ))
+    do
+        if [[ $(git branch -a | grep ${arrayDefaultBranches[x]} | wc -l) -gt 0 ]]
+        then
+             CheckoutBranchAndPull ${arrayDefaultBranches[x]}
+             flagRequiredBranchFound=1
+        fi
+    done
 }
-function FindsWhatBranchRepositoryIsOn() {
-	if [[ "$branch_option" = "* $branch_I_want" ]] #note "* " is used to show branch selection
-	then
-	 #echo "On the correct branch"
-	 flagWhatBranchOn="$flagOnCorrectBranch"
-	elif [[ "$branch_option" = "* $branch_I_want_dev" ]]
-	then
-	 #echo "On the correct branch"
-	 flagWhatBranchOn="$flagOnCorrectBranch"
-	elif [[ "$branch_option" = "$branch_I_want" ]]
-	then
-	 #branch is there but we are not on it
-	 flagWhatBranchOn="$flagOnBranchMaster"
-	elif [[ "$branch_option" = "$branch_I_want_dev" ]]
-	then
-	 #branch is there but we are not on it
-	 flagWhatBranchOn="$FlagOnBranchDev"
-	fi
+
+function CheckoutBranchAndPull(){
+    git stash # saves changes but does not commit before changing branch
+    git checkout ${1}
+    git pull
 }
+
 function MoveToCorrectBranch(){
-	if [[ "$flagWhatBranchOn" -eq "$flagDontKnowBranch" ]]
-	then
-	  echo "######################################################"
-	  echo "# Could NOT find $branch_I_want or $branch_I_want_dev branches              #"
-	  echo "# Please check that you are on the correct branch     #"
-	  echo "# Will run update on current branch                  #"
-	  echo "######################################################"
-	elif [[ "$flagWhatBranchOn" -eq "$flagOnCorrectBranch" ]]
-	then
-	  echo "On the correct branch"
-	else
-	 echo "On wrong branch will try and change- please note will run git stash command"
-
-		if [[ "$flagWhatBranchOn" -eq "$flagOnBranchMaster" ]]
-		then
-		 git stash # saves changes but does not commit before changing branch
-		 git checkout $branch_I_want
-		elif [[ "$flagWhatBranchOn" -eq "$FlagOnBranchDev" ]]
-		then
-		 git stash # saves changes but does not commit before changing branch
-		 git checkout $branch_I_want_dev
-		fi
-	fi
-}
 
+    local flagRequiredBranchFound=0
+
+    if [[ ${1} -gt 0 ]]
+    then
+        echo "Requested branch found: "${2}
+        CheckoutBranchAndPull ${2}
+        local flagRequiredBranchFound=1
+    else
+        echo "Moving to Default Branch"
+        MoveToDefaultBranches
+    fi
+
+    if [[ ${flagRequiredBranchFound} -eq 0 ]]
+    then
+        echo "WARNING NO BRANCHES FOUND"
+    fi
+}
 
-function BranchChecking(){
-	#constants
-	flagDontKnowBranch=0
-	flagOnCorrectBranch=1
-	flagOnBranchMaster=2
-	FlagOnBranchDev=3
 
+function DoesBranchExistForParameterPassedIntoScript(){
+    #if used to check if a parameter is passed into script
+    #this is used to prevent error
+    if [[ ${#1} -eq 0 ]]  #${#var}  Use the length of var.
+    then
+        local branchExist=0
```

Comment:
can you make the variable function scoped e.g. local branchExist at the top of the function